packages/url-utils/src/utils/build-early-exit-match.ts

```diff
@@ -16,14 +21,14 @@ function escapeRegExp(string) {
  * @param {boolean} [options.ignoreProtocol=true] - Whether to strip protocol from URLs
  * @returns {string|null} Regex pattern matching any configured base URL, or null if none configured
  */
-function buildEarlyExitMatch(siteUrl, options = {}) {
+function buildEarlyExitMatch(siteUrl: string, options: BuildEarlyExitMatchOptions = {}): string | null {
     const candidates = [siteUrl, options.imageBaseUrl, options.filesBaseUrl, options.mediaBaseUrl]
-        .filter(Boolean)
-        .map((value) => {
+        .filter((value): value is string => Boolean(value))
+        .map((value: string) => {
             let normalized = options.ignoreProtocol ? value.replace(/http:|https:/, '') : value;
             return normalized.replace(/\/$/, '');
         })
-        .filter(Boolean)
+        .filter((value): value is string => Boolean(value))
```

Comment:
I think in this case we can just do `(value: string): boolean => Boolean(value)`